bridge-fungible-token/src/utils.sw

```diff
@@ -153,15 +153,45 @@ pub fn parse_message_data(msg_idx: u8) -> MessageData {
     let mut msg_data = MessageData {
         token: ZERO_B256,
         from: ZERO_B256,
-        to: Address::from(ZERO_B256),
+        to: Identity::Address(Address::from(ZERO_B256)),
         amount: ZERO_B256,
+        deposit_to_contract: false,
+        len: 160,
     };
 
     // Parse the message data
-    msg_data.token = input_message_data(msg_idx, 32).into();
-    msg_data.from = input_message_data(msg_idx, 32 + 32).into();
-    msg_data.to = Address::from(input_message_data(msg_idx, 32 + 32 + 32).into());
-    msg_data.amount = input_message_data(msg_idx, 32 + 32 + 32 + 32).into();
+    let token = input_message_data(msg_idx, 32);
+    let ptr = __addr_of(msg_data.token);
+    token.buf.ptr().copy_to::<b256>(ptr, 1);
+    // msg_data.token = input_message_data(msg_idx, 32).into();
+    let from = input_message_data(msg_idx, 32 + 32);
+    let ptr_2 = __addr_of(msg_data.from);
+    from.buf.ptr().copy_to::<b256>(ptr_2, 1);
+    // msg_data.from = input_message_data(msg_idx, 32 + 32).into();
+    let amount = input_message_data(msg_idx, 32 + 32 + 32 + 32);
+    let ptr_3 = __addr_of(msg_data.amount);
+    amount.buf.ptr().copy_to::<b256>(ptr_3, 1);
+    // msg_data.amount = input_message_data(msg_idx, 32 + 32 + 32 + 32).into();
+
+    // any data beyond 160 bytes means deposit is meant for a contract.
+    // if data is > 161 bytes, then we also need to call process_message on the destination contract.
+    msg_data.len = input_message_data_length(msg_idx);
+    if msg_data.len > 160u16 {
+        msg_data.deposit_to_contract = true;
+    };
+
+    let mut raw_id = ZERO_B256;
+    let to = input_message_data(msg_idx, 32 + 32 + 32);
+    let ptr_4 = __addr_of(raw_id);
+    to.buf.ptr().copy_to::<b256>(ptr_4, 1);
+
+    msg_data.to = match msg_data.deposit_to_contract {
```

Comment:
might be cleaner to just set `to` in the same if-else block you were using to flag `deposit_to_contract`.

Reply:
Agreed, and done.